src/compiler/checker.ts

```diff
@@ -15336,7 +15336,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                 const type = signature.declaration && getEffectiveReturnTypeNode(signature.declaration);
                 let jsdocPredicate: TypePredicate | undefined;
                 if (!type) {
-                    const jsdocSignature = getSignatureOfTypeTag(signature.declaration!);
+                    const jsdocSignature = getSignatureOfTypeTag(signature.declaration!, /*allowMembers*/ false);

```

Comment:
I can't figure out a test case for which this would matter but I don't see a reason why this would have to **not** allow members.